core/src/main/spark-2.4/org/apache/spark/sql/catalyst/catalog/TiCompositeSessionCatalog.scala

```diff
@@ -60,15 +60,29 @@ class TiCompositeSessionCatalog(val tiContext: TiContext)
     override val sessionCatalog: SessionCatalog = tiContext.sessionCatalog
   }
 
-  val policy: CompositeCatalogPolicy = TiLegacyPolicy(tiContext)
+  lazy val policy: CompositeCatalogPolicy = TiLegacyPolicy(tiContext)
```

Comment:
Previous implementation causes tiContext to be initialized during construction despite it is marked as `lazy`